build.cake

```diff
@@ -15,67 +18,90 @@ Task("Clean")
 
 Task("Restore")
     .Does(() => {
-        NuGetRestore("./src/Shouldly.sln");
+        DotNetCoreRestore("./src/Shouldly.sln", new DotNetCoreRestoreSettings{
+            Verbosity = DotNetCoreVerbosity.Minimal,
+        });
     });
 
 GitVersion versionInfo = null;
+DotNetCoreMSBuildSettings msBuildSettings = null;
+
 Task("Version")
     .Does(() => {
         GitVersion(new GitVersionSettings{
             UpdateAssemblyInfo = false,
             OutputType = GitVersionOutput.BuildServer
         });
         versionInfo = GitVersion(new GitVersionSettings{ OutputType = GitVersionOutput.Json });
-        // Update project.json
-        var updatedProjectJson = System.IO.File.ReadAllText(shouldlyProj)
-            .Replace("1.0.0-*", versionInfo.NuGetVersion);
-
-        System.IO.File.WriteAllText(shouldlyProj, updatedProjectJson);
+        
+        msBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("Version", versionInfo.NuGetVersion)
+                            .WithProperty("AssemblyVersion", versionInfo.AssemblySemVer)
+                            .WithProperty("FileVersion", versionInfo.NuGetVersion);
```

Comment:
What should FileVersion actually be?
Look in the build output at `Adding Environment Variable. name='GitVersion_` for a nice list of options